crowbar_framework/app/models/api/crowbar.rb

```diff
@@ -150,6 +150,24 @@ def compute_resources_available?
       true
     end
 
+    # Check for presence of HA setup, which is a requirement for non-disruptive upgrade
+    def ha_deployed?
+      unless addon_installed? "ha"
+        errors.add(:base, I18n.t("api.crowbar.ha_not_installed"))
```

Comment:
I was expecting we're gonna read the errors somehow when constructing that pre-check map. Or do you think whole `ha_deployed` method should be changed to return the hash instead?